XenAdmin/TabPages/GeneralTabPage.cs

```diff
@@ -1309,38 +1308,51 @@ private void generateGeneralBox()
-                    if (Helpers.ClearwaterOrGreater(p.Connection))
-                        s.AddEntry(Messages.NUMBER_OF_SOCKETS, p.CpuSockets.ToString());
+                s.AddEntry(Messages.POOL_LICENSE, p.LicenseString);
+                if (Helpers.ClearwaterOrGreater(p.Connection))
+                    s.AddEntry(Messages.NUMBER_OF_SOCKETS, p.CpuSockets.ToString());
 
-                    var master = p.Connection.Resolve(p.master);
-                    if (master != null)
+                var master = p.Connection.Resolve(p.master);
+                if (master != null)
+                {
+                    if (p.IsPoolFullyUpgraded)
                     {
-                        if (p.IsPoolFullyUpgraded)
-                        {
-                            s.AddEntry(Messages.SOFTWARE_VERSION_PRODUCT_VERSION, master.ProductVersionText);
-                        }
-                        else
-                        {
-                            var cmd = new RollingUpgradeCommand(Program.MainWindow);
-                            var runRpuWizard = new ToolStripMenuItem(Messages.ROLLING_POOL_UPGRADE_ELLIPSIS,
-                                                                     null,
-                                                                     (sender, args) => cmd.Execute());
-
-                            s.AddEntryLink(Messages.SOFTWARE_VERSION_PRODUCT_VERSION,
-                                           string.Format(Messages.POOL_VERSIONS_LINK_TEXT, master.ProductVersionText),
-                                           new[] { runRpuWizard },
-                                           cmd);
-                        }
+                        s.AddEntry(Messages.SOFTWARE_VERSION_PRODUCT_VERSION, master.ProductVersionText);
+                    }
+                    else
+                    {
+                        var cmd = new RollingUpgradeCommand(Program.MainWindow);
+                        var runRpuWizard = new ToolStripMenuItem(Messages.ROLLING_POOL_UPGRADE_ELLIPSIS,
+                            null,
+                            (sender, args) => cmd.Execute());
+
+                        s.AddEntryLink(Messages.SOFTWARE_VERSION_PRODUCT_VERSION,
+                            string.Format(Messages.POOL_VERSIONS_LINK_TEXT, master.ProductVersionText),
+                            new[] {runRpuWizard},
+                            cmd);
                     }
                 }
             }
 
+            VDI vdi = xenObject as VDI;
+            if (vdi != null)
+            {
+                s.AddEntry(Messages.SIZE, vdi.SizeText,
+                    new PropertiesToolStripMenuItem(new VdiEditSizeLocationCommand(Program.MainWindow, xenObject)));
+
+                SR vdiSr = vdi.Connection.Resolve(vdi.SR);
+                if (vdiSr != null && !vdiSr.IsToolsSR)
+                    s.AddEntry(Messages.DATATYPE_STORAGE, vdiSr.Name);

```

Comment:
Actually the SR.Name does give the host information, but since you mentioned it I'll have a closer look to see how the one property is dfferent from the other, we seem to have too many properties dealing with names and locations...